### ISSUE TITLE
[Suggestion] More config entries for the Chaos Guardian

### ISSUE THREAD

**Issue by BlackjackMKV** [brandon3055/Draconic-Evolution#1415]
I run a server that uses Draconic Evolution among others, and I keep receiving complaints about the Chaos Guardian. I get that you want it to be super hard to kill, but I would greatly appreciate it if we had the ability to disable certain attacks such as the little red homing ones via the config.

Additionally, the option to both make the damage not ignore armors that it shouldn't and make stuff like the Swiftwolf's Rending Gale from ProjectE reflect the projectiles as intended would also be greatly appreciated, as that forms a large number of the complaints we receive due to DE ignoring anything that isn't from itself due to nonstandard code.

Once again, I get it, it is supposed to be super hard even for skilled players. However, not everyone and, indeed, most players can't deal with it. This is why I'm asking for the config settings rather than a flat-out change. That way, the hardcore players can play it their way, and servers meant for more casual players or a wider audience can also be able to do something about it. My only current solution is to place the items you'd receive from beating it at spawn in an infinite barrel so the players don't have to if they don't want to, or else that entire section of the mod is completely blocked off to them.

**Comment by Tahakk** [comment]
it is supposed to be hard, but you can cheese it. The little red and blue projectiles can be shot and destroyed, you can have multiple Capacitors so you can take more critical DMG. For sure maxed out Draconic Armor + Bow is must if you having problems

**Comment by BlackjackMKV** [comment]
It's not so much that I have issues with it, although it is tricky. My issue is more of the complaints I get day in and day out about it. Unfortunately, it seems not everyone can deal with the incoming projectiles as well as they really should. Plus, I keep getting 'bug reports' about all the other armors, weapons, and gadgets from other mods that SHOULD work and don't against this. Gem Armor and Quantom Armor being the two biggest offenders, as they should functionally provide invincibility. Hence the request. >_<

**Comment by Tahakk** [comment]
Even Creative mode wasn't save from the wrath of Chaos Guardian a while back, it's just a way he is done. He is doing chaos DMG and so if you don't have Draconic Armor with the shield active, well that's the problem. Anyway, you can always do it in more than one player if having problems

Please note that if I say 'you' I mean it globally.

**Comment by BlackjackMKV** [comment]
Sure, but I've already told the players that. Like I said earlier, my server's playerbase is more casuals who, although willing to challenge something hard every now and again, don't want to have to organize a raid every time they want to take on of those things on. Tbh, it's kinda gruesome to watch. Usually there'll be about ten of them at a time who take turns respawning and dashing back from the waypoints(teleports from a different mod) they have set to keep the fight constantly going. I just want to make it so they could solo it with some effort but without the super high amount they would normally. Hence why I want to disable a attack pattern or two and/or allow other mods' equipment to work such as Swiftwolf's Rending Gale(PE), Gem Armor(PE), IC2 Quantom Armor, Modular Power Armor, Elementium Armor(Botania) and so on.

Long story short, I told them I'd see if I could do SOMETHING about it and in the meantime set up a custom recipe identical to the one in Sky Odyssey to fusion craft the chaos crystals(20 Awakened cores with a dark matter slotted in for a draconic level altar, and 10 cores with the dark matter for the chaotic level altars). It works, but completely removes any need to fight the boss at all. I hate to cut out such a nicely built boss fight simply because I lack the config lines to make it a tad easier.

**Comment by Tahakk** [comment]
As an server admin I feel your pain. Although more configs never hurt anybody (:

**Comment by BlackjackMKV** [comment]
Agreed. 'Tis why I thought to suggest it. Worst case scenario, the answer is no and the boss stays completely optional on my server. It'd sadden me a bit, since I do love the aesthetic and the whole design of the fight's mechanics, but you've gotta keep your players happy somehow. Best case scenario, the answer is yes, it gets implemented, and then I update the mod so quick that even Sonic is gonna be asking what the hell just shot past. :P

**Comment by FoxMcloud5655** [comment]
As the current pandemic sweeping around the nation allows more and more people to play Minecraft than ever before, we are getting an influx of people who are a completely different playerbase than we had when this boss was first implemented.

There have been NUMEROUS complaints in these issues (if you did your research) about the Chaos Guardian being too hard.  Personally, I take all of the shield values for Wyvern/Draconic Armor and divide them by 4 in my config file, as I think it's too powerful.  That aside, I do realize that at least half of the players out there have trouble fighting this boss.

Adding more config options never hurt, so I might make a pull request for this.  If I do, I can update this issue with an unofficial fix if you'd like.

**Comment by BlackjackMKV** [comment]
Yep. I looked around a ton for a possible fix of some kind before coming here. It's not unusual for me to add mods to fix another mod that change a completely different mod even. No dice though. I would totally take you up on that pull req though if you're willing. If you don't have the time or it turns out to be trickier than you wanted, that's okay too though. Not like I'm going to try to force you to do something for free if it isn't worth it to you. :P

**Comment by Unknowcool** [comment]
I have been trying to reduce the attack of the Chaos Guardian myself lately. It's a surprise to have stumbled on this while browsing. I like the whole appeal behind the mod, and the Chaos Guardian is a nice addition for the ones daunting enough to travel further than the initial island one finds when entering the End.

The only issue I've had with it is how the mod needs you to use their armor. It's not a problem if you have a modpack focusing entirely on DE since that's going to be the go point. However, if DE's more of a side mod, that can be a problem in some way. I find it interesting to have the ability to lets say, partake in gear from other mods and be creative during the Chaos Guardian fight. (Looking at you magic mods)

The suggestion to add configs to edit the stats of the Chaos Guardian is without a doubt an interesting one to look at since I believe we can tweak all of the different armor, tool and weapon pieces that this mod offers.

**Comment by BlackjackMKV** [comment]
Since it seems like this fell under the radar somehow, I'm going to go ahead and necro this a little bit since my request is still valid as far as I can tell. Ran through all the configs for the most recent version and it's still not a thing, provided I didn't somehow miss it.

**Comment by FoxMcloud5655** [comment]
You're correct, and I think it's a nice idea, but there's probably a reason why Brandon hasn't done it.

**Comment by BlackjackMKV** [comment]
> You're correct, and I think it's a nice idea, but there's probably a reason why Brandon hasn't done it.

Perhaps, though admittedly I'd still like to know why. It'd be a massive help, especially when he's made it kill even _creative_ players. Why do that at all? It's not like it stops anyone from cheating in the first place since they could just give themselves the armor with commands if they can go creative. And sure, stuff like ProjectE's Red Matter Armor can confer total invulnerability and trivialize the fight, but, at that point, why even bother with the chaos guardians if they're just able to instantly kill you for existing regardless of your gear? It's not like people will switch to using DE's stuff to fight them; they'll just ignore the chaos guardians entirely since their gear is still functional everywhere else in the game, right?

Making people use your stuff by force just doesn't really work, and mostly just renders a lot of mods borderline incompatible while reducing the amount of use a mod can have in a modpack. Hence why I want the option so badly. I'd love to make a modpack that has multiple avenues for being endgame, rather than everyone being forced to use DE or avoid large swaths of the End.
